support-modules/supporter-product-data-dynamo/src/main/scala/com/gu/supporterdata/services/SupporterDataDynamoService.scala

```diff
@@ -32,33 +32,49 @@ class SupporterDataDynamoService(client: DynamoDbAsyncClient, tableName: String)
     // Dynamo will delete expired subs at the start of the day, whereas the subscription actually lasts until the end of the day
     val expiryDate = item.termEndDate.plusDays(1)
     val expiryDateName = "expiryDate"
+    val contributionAmount = "contributionAmount"
+    val contributionCurrency = "contributionCurrency"
 
     val key = Map(
       identityId -> AttributeValue.builder.s(beneficiaryIdentityId).build,
       subscriptionName -> AttributeValue.builder.s(item.subscriptionName).build,
     ).asJava
 
-    val updateExpression =
+    val nonOptionExpression =
       s"""SET
           $productRatePlanId = :$productRatePlanId,
           $productRatePlanName = :$productRatePlanName,
           $termEndDate = :$termEndDate,
           $contractEffectiveDate = :$contractEffectiveDate,
           $expiryDateName = :$expiryDateName
           """
-    val attributeValues = Map(
+    val updateExpression =
+      if (item.contributionAmount.isDefined)
+        nonOptionExpression + s",\n          $contributionAmount = :$contributionAmount,\n          $contributionCurrency = :$contributionCurrency"
+      else nonOptionExpression
```

Comment:
what is nonOption as it's not clear the meaning? is it only the required values for all products?
how about setRequiredValuesExpression.

also would it be clearer to separate the defining the parts from combining them, rather than accumulating as we go along?
Something like this...
```scala
    val setRequiredValuesExpression =
      s"""SET
          $productRatePlanId = :$productRatePlanId,
          $productRatePlanName = :$productRatePlanName,
          $termEndDate = :$termEndDate,
          $contractEffectiveDate = :$contractEffectiveDate,
          $expiryDateName = :$expiryDateName
          """

    val additionalFieldExpressions =
      if (item.contributionAmount.isDefined)
        List(
          s"$contributionAmount = :$contributionAmount",
          s"$contributionCurrency = :$contributionCurrency",
        )
      else Nil

    val updateExpression =
      (setRequiredValuesExpression ++ additionalFieldsExpression)
        .mkString(s",\n          ")
```